src/js/core/methods/EthereumSignTypedData.js

```diff
@@ -0,0 +1,172 @@
+/* @flow */
+
+import AbstractMethod from './AbstractMethod';
+import { validateParams, getFirmwareRange } from './helpers/paramsValidator';
+import { validatePath } from '../../utils/pathUtils';
+import { getEthereumNetwork } from '../../data/CoinInfo';
+import { toChecksumAddress, getNetworkLabel } from '../../utils/ethereumUtils';
+import type { CoreMessage, EthereumNetworkInfo } from '../../types';
+import type {
+    MessageResponse,
+    EthereumTypedDataSignature,
+    EthereumTypedDataStructAck,
+    EthereumTypedDataValueRequest,
+} from '../../types/trezor/protobuf';
+import { ERRORS } from '../../constants';
+import type { EthereumSignTypedData as EthereumSignTypedDataParams } from '../../types/networks/ethereum';
+import { getFieldType, parseArrayType, encodeData } from './helpers/ethereumSignTypedData';
+
+type Params = {
+    ...EthereumSignTypedDataParams,
+    path: number[],
+    network?: EthereumNetworkInfo,
+};
+
+export default class EthereumSignTypedData extends AbstractMethod {
+    params: Params;
+
+    constructor(message: CoreMessage) {
+        super(message);
+
+        this.requiredPermissions = ['read', 'write'];
+
+        const { payload } = message;
+
+        // validate incoming parameters
+        validateParams(payload, [
+            { name: 'path', obligatory: true },
+            { name: 'data', type: 'object', obligatory: true },
+            { name: 'metamaskV4Compatibility', type: 'boolean', obligatory: true },
```

Comment:
Recently we have settled that we would like to keep public interface on connect as similar to protobuf messages as possible - to be specific here, we want to use snake case for parameters that are part of protobuf so that we don't have to tranform between various object shapes. Please change `metamaskV4Compatibility` to `metamask_v4_compat`

In **future** we would also change `path` to `address_n`, but **not** in this PR just to stay consistent with other methods.

Reply:
:+1: Done in https://github.com/trezor/connect/pull/983/commits/55f3912007695f0fcb324b7ca41aac00ffcfc92e